### PR TITLE
DR-1339: clear IAM policies from perf

### PR DESCRIPTION
**Pull request by snf2ye**
## Goal
We run into a IAM policy limit w/ glcoud if we have more than 250 IAM policies. On every test run, policies are created in the data project. We can clear them out before the test suite runs to maintain an acceptable number of policies. 

## In Action
Successful run [here](https://github.com/DataBiosphere/jade-data-repo/runs/1178410354?check_suite_focus=true#step:4:1).

### CODE REVIEW

File: .github/workflows/test-runner-on-perf.yml [hunk 1]
```diff
@@ -8,14 +8,40 @@ env:
   GOOGLE_ZONE: us-central1
   K8_CLUSTER: jade-master-us-central1
 on:
-  schedule:
-    - cron: '0 4 * * *' # run at 4 AM UTC, 12PM EST.
+  #schedule:
+  #  - cron: '0 4 * * *' # run at 4 AM UTC, 12PM EST.
```
Comment (nmalfroy) [inline .github/workflows/test-runner-on-perf.yml]: Did you mean to leave these comments?

**Review by nmalfroy — APPROVED** [review]
lgtm

File: .github/workflows/test-runner-on-perf.yml [hunk 2]
```diff
@@ -8,14 +8,40 @@ env:
...
+        uses: hashicorp/vault-action@c8b383ee
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.PERF_ROLE_ID }}
+          secretId: ${{ secrets.PERF_SECRET_ID }}
+          secrets: |
+            secret/dsde/datarepo/perf/datarepo-api-sa key | B64_APPLICATION_CREDENTIALS ;
+            secret/dsde/datarepo/perf/datarepo-sql-db datarepopassword | DATAREPO_PASS ;
+            secret/dsde/datarepo/perf/datarepo-sql-db stairwaypassword | STAIRWAY_PASS ;
```
Comment (fboulnois) [inline .github/workflows/test-runner-on-perf.yml]:
```suggestion
```

These aren't necessary.

File: .github/workflows/test-runner-on-perf.yml [hunk 3]
```diff
@@ -8,14 +8,40 @@ env:
...
+  #  - cron: '0 4 * * *' # run at 4 AM UTC, 12PM EST.
+  workflow_dispatch: {}
 jobs:
   test-runner-perf:
     runs-on: ubuntu-latest
     steps:
       - name: "Checkout code"
         uses: actions/checkout@v2
+      - name: "Import Vault perf secrets"
+        uses: hashicorp/vault-action@c8b383ee
```
Comment (fboulnois) [inline .github/workflows/test-runner-on-perf.yml]:
```suggestion
        uses: hashicorp/vault-action@v2.0.1
```

Hashicorp stabilized their vault action.

**Review by fboulnois — CHANGES_REQUESTED** [review]
A few suggestions:

File: .github/workflows/test-runner-on-perf.yml (hunk 2 above)
Comment (marikomedlock) [inline .github/workflows/test-runner-on-perf.yml]: Do you need the DB passwords?

**Review by marikomedlock — APPROVED** [review]
LGTM. One comment inline (possibly the same as Florian's).

Out of scope for this PR, but is there a way for us to share code between the GH actions? I understand that we can do that with scripts in the datarepo-actions repo. I also understand that we are trying to move away from that model because it requires two tightly coordinated merges into different GH repositories. So, I'm wondering if we can use scripts that live in the main jade-data-repo? It seems like we will/already have a lot of shared code between actions.

File: .github/workflows/test-runner-on-perf.yml (hunk 1 above)
Comment (snf2ye) [reply .github/workflows/test-runner-on-perf.yml]: Yes, I'm going to switch back to the schedule as soon as I'm finished testing. I am tracking a clean-up list before merging in the [PR Description](https://github.com/DataBiosphere/jade-data-repo/pull/637#issue-494321344).

File: .github/workflows/test-runner-on-perf.yml (hunk 2 above)
Comment (snf2ye) [reply .github/workflows/test-runner-on-perf.yml]: I do not! Removed.

**Review by fboulnois — CHANGES_REQUESTED** [review]
I think the rebase might have discarded my suggestions (vault action version and db passwords).

**Comment by snf2ye** [comment]
> I think the rebase might have discarded my suggestions (vault action version and db passwords).

Oh, good catch. I missed that.

**Review by fboulnois — APPROVED** [review]
latest changes lgtm

**Pull request merged**